modules/ctrl_dbus/ctrl_dbus.c

```diff
@@ -374,8 +409,14 @@ on_name_acquired(GDBusConnection *connection, const gchar *name, gpointer arg)
 	}
 
 	info("ctrl_dbus: dbus interface %s exported\n", name);
-	module_event("ctrl_dbus", "exported", NULL, NULL,
-			"dbus interface %s exported", name);
+
+	s = str_len(name) + sizeof(fmt);
+	modev = mem_zalloc(sizeof(*modev), modev_destructor);
+	modev->txt = mem_zalloc(s, NULL);
+	str_dup(&modev->event, "exported");
+	re_snprintf(modev->txt, s, fmt, name);
```

Comment:
this code can be written in a simpler way:

```c
err = re_sdprintf(&modev->txt, "asd asd asd %d", 42);
```

also please check return value of functions